fabtests/multinode/src/core.c

```diff
@@ -372,17 +389,19 @@ int send_recv_barrier(int sync)
 	int ret, i;
 
 	for (i = 0; i < pm_job.num_ranks; i++) {
-		ret = ft_post_rx_buf(ep, opts.transfer_size,
-			     	     &barrier_rx_ctx[i],
-			     	     rx_buf, mr_desc, 0);
+		remote_fi_addr = pm_job.fi_addrs[i];
+		ret = ft_post_rx_buf(ep, FT_MAX_CTRL_MSG,
+				     &barrier_rx_ctx[i],
+				     rx_barrier + (FT_MAX_CTRL_MSG * i),
+				     barrier_desc, BARRIER_TAG);
```

Comment:
Using a 0-byte message, we don't need the barrier context or buffers

Reply:
updated

Reply:
We still don't need to post a buffer -- that can be NULL, size 0, and no memory descriptor needed.